### DISCUSSION TITLE
Weblate Formats with Docker Image

### DISCUSSION THREAD

**Discussion by benydc** [WeblateOrg/weblate#12999] [Q&A]
Hello,

I'm using the Weblate with Docker image deployed for managing translations and would like to add support for a new file format, specifically for Subtitles (e.g., .srt files).

Here’s what I've tried so far:

- I have checked the available formats in Weblate component but couldn’t find an option for subtitles out of the box.
- I’m aware that Weblate supports various format .srt
- I’ve looked into customizing formats, but I’m not sure how to integrate it properly within the Docker environment, there is no environment variable?

Thank you!

**Comment by sszczep** [comment]
Hey @benydc, have you solved your issue by any chance?

**Reply by benydc** [reply]
@sszczep Yes, it was actually when importing a zip file that I could select the SubRip format. However, we stopped using it because the translations were somehow getting messed up in the subtitles.

**Reply by sszczep** [reply]
I see. I was hoping you had to develop your own custom format parser, because that's what I need to do. Thanks anyway

**Reply by nijel** [reply]
You can place custom code in `/app/data/python/` and then use it in [WEBLATE_FORMATS](https://docs.weblate.org/en/latest/admin/config.html#weblate-formats) in /app/data/settings-override.py (see https://docs.weblate.org/en/latest/admin/install/docker.html#overriding-settings-from-the-data-volume).

Or contribute the file format implementation and will land in the next release :wink:.

**Reply by sszczep** [reply]
Hey @nijel, thank you for your reply. I've tried that solution however I couldn't get weblate to register the new format. It doesn't matter now - since I'm using REST for uploading/downloading translations, I created a bidirectional XLIFF mapper in the same tool.

As for adding that format to official repository - it's our internal format and wouldn't be a much of a use for others ;)

**Reply by nijel** [reply]
The format needs unique `format_id`, then adding the class name to `WEBLATE_FORMATS` in `/app/data/settings-override.py` and restarting container should work.

**Reply by sszczep** [reply]
I'll maybe give it a shot in the future, for now current solutions suits us well. Thank you for your help.
